src/core/queries.ml

```diff
@@ -50,45 +53,55 @@ let handle_query : Sig_state.t -> proof_state option -> p_query -> unit =
           | Some [] ->
               if Eval.eq_modulo ctxt t u = must_fail then
                 fatal q.pos "Assertion failed."
-      end
+      end;
+      None
   | P_query_debug(e,s) ->
       (* Just update the option, state not modified. *)
       Console.set_debug e s;
-      out 3 "(flag) debug → %s%s\n" (if e then "+" else "-") s
+      out 3 "(flag) debug → %s%s\n" (if e then "+" else "-") s;
+      None
   | P_query_verbose(i) ->
       (* Just update the option, state not modified. *)
       if Timed.(!Console.verbose) = 0 then
         (Timed.(Console.verbose := i); out 1 "(flag) verbose → %i\n" i)
       else
-        (out 1 "(flag) verbose → %i\n" i; Timed.(Console.verbose := i))
+        (out 1 "(flag) verbose → %i\n" i; Timed.(Console.verbose := i));
+      None
   | P_query_flag(id,b) ->
       (* We set the value of the flag, if it exists. *)
       (try Console.set_flag id b
        with Not_found -> wrn q.pos "Unknown flag \"%s\"." id);
-      out 3 "(flag) %s → %b\n" id b
+      out 3 "(flag) %s → %b\n" id b;
+      None
   | P_query_infer(pt, cfg) ->
       let env = Proof.focus_env ps in
       let ctxt = Env.to_ctxt env in
       let scope = Scope.scope_term Public ss env in
       let t = scope pt in
       let a = Infer.infer Unif.solve_noexn pt.pos ctxt t in
-      out 1 "(infr) %a : %a\n" pp_term t pp_term (Eval.eval cfg ctxt a)
+      out 1 "(infr) %a : %a\n" pp_term t pp_term (Eval.eval cfg ctxt a);
+      Some (Format.asprintf "%a : %a" pp_term t pp_term
+              (Eval.eval cfg ctxt a))
```

Comment:
It would be better to factorize the result as follows: `let res = Format.asprintf ... in out 1 "..."; Some res`. Idem in the other queries.

Reply:
No sorry. I mean the computations only: `let a = Eval.eval cfg ctxt a in ...`